src/back/wgsl/writer.rs

```diff
@@ -461,19 +568,49 @@ impl<W: Write> Writer<W> {
             }
             Statement::Return { value } => {
                 write!(self.out, "{}", INDENT.repeat(indent))?;
+                write!(self.out, "return")?;
                 if let Some(return_value) = value {
-                    write!(self.out, "return ")?;
+                    // The leading space is important
+                    write!(self.out, " ")?;
                     self.write_expr(module, return_value, &func_ctx)?;
-                    writeln!(self.out, ";")?;
-                } else {
-                    writeln!(self.out, "return;")?;
                 }
+                writeln!(self.out, ";")?;
             }
             // TODO: copy-paste from glsl-out
             Statement::Kill => {
                 write!(self.out, "{}", INDENT.repeat(indent))?;
                 writeln!(self.out, "discard;")?
             }
+            // TODO: copy-paste from glsl-out
+            Statement::Store { pointer, value } => {
+                write!(self.out, "{}", INDENT.repeat(indent))?;
+                self.write_expr(module, pointer, func_ctx)?;
+                write!(self.out, " = ")?;
+                self.write_expr(module, value, func_ctx)?;
+                writeln!(self.out, ";")?
+            }
+            crate::Statement::Call {
+                function,
+                ref arguments,
+                result,
+            } => {
+                write!(self.out, "{}", INDENT.repeat(indent))?;
+                if let Some(expr) = result {
+                    self.start_baking_expr(module, expr, &func_ctx)?;
+                    self.named_expressions.insert(expr.index());
+                }
+                let func_name = &self.names[&NameKey::Function(function)];
+                write!(self.out, "{}(", func_name)?;
+                for (index, argument) in arguments.iter().enumerate() {
+                    self.write_expr(module, *argument, func_ctx)?;
+                    // Only write a comma if isn't the last element
+                    if index != arguments.len().saturating_sub(1) {
```

Comment:
WGSL accepts trailing commas, so this can be simplified a but

Reply:
TIL. Thank you! But here and in other places with the same logic, we also add spaces between arguments.